### ISSUE TITLE
Blood freeze on Win 98

### ISSUE THREAD

**Issue by AITUS95** [joncampbell123/dosbox-x#3768]
### Describe the bug

Hi, a few days ago I opened a report similar to this problem ([here](https://github.com/joncampbell123/dosbox-x/issues/3752)) but on MS-DOS with Dosbox-x, now I'm on Windows 98 with Dosbox-x, the error comes from the ATAPI device as the log describes:

![pic](https://user-images.githubusercontent.com/48457684/193400200-cc55ec38-5534-4093-921a-5ae715574c56.PNG)

On MS-DOS it works by changing the ide driver of the cdrom, but on windows 98 it is not possible since the driver seems to be stuck on the Dosbox one ... any solution?

### Steps to reproduce the behaviour

1. Run Blood 
2. New game
3. Black screen

### What operating system(s) this bug have occurred on?

Windows 10 Pro x64

### What version(s) of DOSBox-X have this bug?

v0.84.3, Windows SDL2, 64-bit

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by rderooy** [comment]
Actually in Win98 the driver is provided by Win98, not by DOSBox-X.

**Comment by AITUS95** [comment]
> Actually in Win98 the driver is provided by Win98, not by DOSBox-X.

When I try to change the cdrom driver via config.sys, I always get the following device:

![guest os_000](https://user-images.githubusercontent.com/48457684/193419871-6a2c5ac5-89d8-451c-bde8-95868205cd37.png)

I think it should detect another type of CDROM, on msdos it works successfully.

**Comment by rderooy** [comment]
That is the virtual CD-ROM **device**. As I mentioned, when you boot a guest OS such as Win98, dosbox-x does not provide a driver. In the case of a Win9x/ME guest OS, the driver used will be the standard IDE/ATAPI driver provided by the guest OS.

**Comment by AITUS95** [comment]
How do I change the guest's IDE / ATAPI driver?

**Comment by maron2000** [comment]
I thought you mentioned that mounting your CD image via Daemon tools installed on Windows 95 guest worked.
It didn't on Windows 98?

**Comment by AITUS95** [comment]
> Pensavo che tu abbia menzionato che il montaggio dell'immagine del CD tramite gli strumenti Daemon installati sul guest di Windows 95 ha funzionato. Non funzionava su Windows 98?

yes, with daemon tools the game works on windows 9x, so confirm the ide drive thing, in any case the game works without music with daemon for missing WDM drivers.

**Comment by grapeli** [comment]
>... any solution?

Very simple. Only use it with the **dos included in dosbox-x**. It doesn't need any special other DOS version - ms-dos, freedos, pc-dos, etc. I tested, including 3dfx under dosbox-x. It works.

Blood 1.0 version behaves under windows in dosbox-x as described. I checked.
If you want to play under Windows, patch this game.
Blood 1.11 works.

https://user-images.githubusercontent.com/452325/193431035-464229ac-4c42-4e1d-ac1c-fa4b867da1de.mp4

In this file you have a list of the issues that the Blood update brings.
[PATCH.TXT](https://github.com/joncampbell123/dosbox-x/files/9691319/PATCH.TXT)

Search for a special driver? What for?

**Comment by AITUS95** [comment]
> 

I expressed myself badly, I was looking for a solution not an alternative, in fact I could have played blood 1.0 on my real hardware with windows 98, only I wanted to report it here as a bug because I like to have dosbox-x as accurate as possible.  I often compare with real hardware and emulators and always look for a solution.

**Comment by maron2000** [comment]
The root cause of this phenomena is that the game issues an illegal ATAPI command. So I don't think it is worth hacking DOSbox-X while we have a clear workaround.

**Comment by grapeli** [comment]
> 
>...in fact I could have played blood 1.0 on my real hardware with windows 98...

It is possible that with a different hardware configuration you could be less fortunate. The fact that this game is functioning properly in version 1.11 indicates that it could be a bug on her side.

For version 1.0, there is another way to work around it. To abandon the CD soundtrack in favor of MIDI. Run through `blood -nocd` or don't mount CD image. You can also convert CD Audio track to mp3, vorbis ogg, opus, flac.

**Comment by rderooy** [comment]
If possible of course, DOSBox-X should probably respond to an invalid command the same way as real hardware would. Assuming that error handling in such cases is even consistent with different devices.

But first of all, you need to know how real hardware behaves in such as case

**Comment by AITUS95** [comment]
> > ...in fact I could have played blood 1.0 on my real hardware with windows 98...
> 
> It is possible that with a different hardware configuration you could be less fortunate. The fact that this game is functioning properly in version 1.11 indicates that it could be a bug on her side.
> 
> For version 1.0, there is another way to work around it. To abandon the CD soundtrack in favor of MIDI. Run through `blood -nocd` or don't mount CD image. You can also convert CD Audio track to mp3, vorbis ogg, opus, flac.

yes it would be a good alternative, even if I don't know how to play the tracks once converted during the game.

> If possible of course, DOSBox-X should probably respond to an invalid command the same way as real hardware would. Assuming that error handling in such cases is even consistent with different devices.
> 
> But first of all, you need to know how real hardware behaves in such as case

i should look for a log if it exists in the game, i am not at home right now so as soon as i can i will try on real hardware to understand what is the behavior of the game with the disc.

**Comment by grapeli** [comment]
Oh sorry, I will mislead you. Converting CD tracks to another format does not change anything. I still get the same error.
```
IDE abort dh=0xa0 with error on 0x170
Unknown ATAPI command 1b 00 00 00 03 00 00 00 00 00 00 00
```

**Comment by maron2000** [comment]
While this document says that 0x03 is not supported for command 0x1b(START/STOP unit),
https://www.t10.org/ftp/x3t9.2/document.87/87-106r0.txt
This one says that it closes the tray (and spin up?).
https://www.bswd.com/sff8020i.pdf
Maybe we can consider implementing the latter.

**Comment by AITUS95** [comment]
so it would be a tray problem, the implementation would then also have the disk eject function?

**Comment by rderooy** [comment]
No, I think the solution would be for it to just return the equivalent of "ok".

**Comment by grapeli** [comment]
Blood updates are available [here](http://www.r-t-c-m.com/knowledge-base/downloads-rtcm/blood-patches/).
The [first patch](http://www.r-t-c-m.com/knowledge-base/downloads-rtcm/blood-patches/r0001.exe) in the series, updating the game to version v1.01, fixes the whole problem.

**Comment by Torinde** [comment]
> No, I think the solution would be for it to just return the equivalent of "ok".

I think @AITUS95 means that the "mirror"-command "eject CD" can be implemented as well? E.g. an option when the "eject CD" is issued by software - to open a dialog box prompting the user to choose ISO image for mounting?

Regardless of having patched Blood it may be good to implement both CD commands - in case some other software relies on those...

**Comment by AITUS95** [comment]
yes i think so too, i like the idea that dosbox-x has no problem with any version of a game, i prefer it that way.
